### PR TITLE
EEG seizure detection pipeline: feature engineering, ML models, bug fixes, and cleanup

### PR DESCRIPTION
**Pull request by GarunaBTW**
## Summary

- **Full feature engineering pipeline**: `AdvancedFeatureExtractor` producing 528 features per 1-second EEG window (33 per channel x 16 channels) — time-domain, Hjorth, nonlinear, frequency, FFT, wavelet, and cross-channel connectivity features
- **Classical ML models**: LightGBM, XGBoost, and Random Forest with Optuna Bayesian hyperparameter tuning (50 trials, maximising val AUCPR)
- **TabNet**: attention-based tabular model with memory-mapped data pipeline for large-scale training
- **Portable config system**: all YAML paths resolved relative to config file — no hardcoded paths anywhere
- **9 bugs fixed**: NumPy compatibility, single-class metric crash, TabNet sparsity sign, feature column validation, memmap row count, config path resolution, and more
- **Verified**: `check_setup.py` and `stress_test.py` (19/19) pass on synthetic data without real EEG files

## Results (subject-independent evaluation on CHB-MIT test set)

| Model | Test AUCPR | Test ROC-AUC | Test F1 |
|-------|-----------|-------------|---------|
| Random Forest (Optuna) | **0.357** | 0.882 | 0.152 |
| TabNet | 0.309 | 0.856 | **0.220** |
| XGBoost (Optuna) | 0.251 | **0.927** | 0.192 |
| LightGBM (Optuna) | 0.230 | 0.933 | 0.195 |

Primary metric: **AUCPR** — correct for severe class imbalance (0.3% seizure rate).

## Folder structure

```
src/
├── config/                      # YAML configs (all relative paths)
├── feature_engineering/         # AdvancedFeatureExtractor + pipeline
└── models/
    ├── baseline/                # train_model.py, train_tabnet.py
    ├── improved/                # optuna_lightgbm/xgboost/random_forest.py
    └── utils/                   # config, data, metric, io, plot, memmap utils
```

## Test plan

- [ ] Pull branch and run `python check_setup.py` — all checks pass
- [ ] Run `python stress_test.py` — 19/19 pass
- [ ] Confirm no hardcoded paths: `grep -r 'spring-2026-group2' src/`
- [ ] Run feature extraction on one split and confirm output CSV is generated
- [ ] Run `python -m src.models.improved.optuna_lightgbm --config src/config/baseline_lightgbm.yaml` and verify metrics JSON is written

### CODE REVIEW

**Comment by askmy-stack** [comment]
First Push good to go and merge it with the main.
